example/hessian.cpp

```diff
@@ -0,0 +1,259 @@
+// when converted to grayscale,
+// which was previously observed on
+// canny edge detector for test input
+// used for this example
+gil::gray8_image_t to_grayscale(gil::rgb8_view_t original)
+{
+    gil::gray8_image_t output_image(original.dimensions());
+    auto output = gil::view(output_image);
+    constexpr double max_channel_intensity = (std::numeric_limits<std::uint8_t>::max)();
+    for (long int y = 0; y < original.height(); ++y)
+    {
+        for (long int x = 0; x < original.width(); ++x)
+        {
+            // scale the values into range [0, 1] and calculate linear intensity
+            auto& p = original(x, y);
+            double red_intensity = p.at(std::integral_constant<int, 0>{})
+                / max_channel_intensity;
+            double green_intensity = p.at(std::integral_constant<int, 1>{})
+                / max_channel_intensity;
+            double blue_intensity = p.at(std::integral_constant<int, 2>{})
+                / max_channel_intensity;
+            auto linear_luminosity = 0.2126 * red_intensity
+                                    + 0.7152 * green_intensity
+                                    + 0.0722 * blue_intensity;
+
+            // perform gamma adjustment
+            double gamma_compressed_luminosity = 0;
+            if (linear_luminosity < 0.0031308)
+            {
+                gamma_compressed_luminosity = linear_luminosity * 12.92;
+            } else
+            {
+                gamma_compressed_luminosity = 1.055 * std::pow(linear_luminosity, 1 / 2.4) - 0.055;
+            }
+
+            // since now it is scaled, descale it back
+            output(x, y) = gamma_compressed_luminosity * max_channel_intensity;
+        }
+    }
+
+    return output_image;
+}
+
+void apply_gaussian_blur(gil::gray8_view_t input_view, gil::gray8_view_t output_view)
+{
+    constexpr static auto filter_height = 5ull;
+    constexpr static auto filter_width = 5ull;
+    constexpr static double filter[filter_height][filter_width] =
+    {
+        2,  4,  6,  4,  2,
+        4, 9, 12, 9,  4,
+        5, 12, 15, 12,  5,
+        4, 9, 12, 9,  4,
+        2,  4,  5,  4,  2,
+    };
+    constexpr double factor = 1.0 / 159;
+    constexpr double bias = 0.0;
+
+    const auto height = input_view.height();
+    const auto width = input_view.width();
+    for (std::ptrdiff_t x = 0; x < width; ++x) {
+        for (std::ptrdiff_t y = 0; y < height; ++y) {
+            double intensity = 0.0;
+            for (std::ptrdiff_t filter_y = 0; filter_y < filter_height; ++filter_y)
+            {
+                for (std::ptrdiff_t filter_x = 0; filter_x < filter_width; ++filter_x)
+                {
+                    int image_x = x - filter_width / 2 + filter_x;
+                    int image_y = y - filter_height / 2 + filter_y;
+                    if (image_x >= input_view.width() || image_x < 0
+                        || image_y >= input_view.height() || image_y < 0)
```

Comment:
Minor formatting-nit-pick: I'd put the binary operator (`||`) at the and of the previous line, which aligns the two sub-expressions.